Track hostname of running jobs

fredden opened the pull request:
This pull request adds a 'hostname' column to the `cron_schedule` table and records the name of the host running the cron job (like we already do for process identifier). This field is then checked when performing process-related tasks such as marking jobs as "gone away" or killing jobs.

Fixes: #98

fredden committed:
Add hostname column to cron_schedule table

fredden committed:
Show hostname in grid

fredden committed:
Set hostname when running job

fredden committed:
Add integration test to cover setting hostname

fredden committed:
Allow retreival of hostname from data model

fredden committed:
Only mark jobs as 'gone' if on same host

fredden committed:
Only kill jobs on same host

fredden committed:
Remove dependency on /proc filesystem (ie Linux)

fredden committed:
Update integration tests to cover new behaviour

Ethan3600 commented on Model/ProcessManagement.php:
```diff
@@ -9,11 +9,14 @@ class ProcessManagement
     public function isPidAlive(int $pid): bool
     {
-        return \file_exists('/proc/' . $pid);
+        return \posix_kill($pid, 0);
```
TIL: _If sig is 0, then no signal is sent, but error checking is still performed; 
this can be used to check for the existence of a process ID or process 
group ID._

https://unix.stackexchange.com/questions/169898/what-does-kill-0-do

Nice! :+1:

Ethan3600 commented:
Code looks great! There's some failing integration tests (I know there's some weirdness with the 2.4 tests as discussed in another PR). If we can make those go away, I think we can merge this puppy!

fredden commented:
Test failures should go away when #143 and #142 land. I'm happy to wait until they're sorted & merged before progressing with this further.

fredden committed:
Merge remote-tracking branch 'upstream/1.x' into track-cron-hostname

fredden committed:
Correct hostname column type

Ethan3600 commented:
Let's make coveralls happy and I'll make a release!

fredden committed:
Remove unused code

The function this was wrapping was introduced in v2.2.0-RC1.2 and was called
within Magento\Cron\Model\Schedule::tryLockJob(), which we also have an after
plugin for. As we are setting (and now saving, see #142) pid & hostname in
there, this around plugin is no longer required. Since v2.3.5, tryLockJob() no
longer calls this function, further rendering this plugin obsolete.

fredden committed:
Merge remote-tracking branch 'upstream/1.x' into track-cron-hostname
